<launch>

    <!-- start the Realsense -->
    <arg name="serial_no"             default=""/>
    <arg name="json_file_path"        default=""/>
    <arg name="camera"                default="camera"/>

    <!-- color max:1920x1080 depth max:1280x720 -->
    <group ns="$(arg camera)">
        <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="serial_no"         value="$(arg serial_no)"/>
        <arg name="json_file_path"    value="$(arg json_file_path)"/>
        <arg name="depth_width"       value="640"/>
        <arg name="depth_height"      value="480"/>
        <arg name="depth_fps"         value="15"/>
        <arg name="color_width"       value="640"/>
        <arg name="color_height"      value="480"/>
        <arg name="color_fps"         value="15"/>
        <arg name="enable_depth"      value="true"/>
        <arg name="enable_color"      value="true"/>
        <arg name="enable_infra1"     value="false"/>
        <arg name="enable_infra2"     value="false"/>
        <arg name="enable_fisheye"    value="false"/>
        <arg name="enable_gyro"       value="false"/>
        <arg name="enable_accel"      value="false"/>

        <arg name="enable_pointcloud" value="true"/>
        <arg name="align_depth"       value="true"/>
        <arg name="enable_sync"       value="true"/>
        <arg name="tf_prefix"         value="$(arg camera)"/>

        <arg name="clip_distance"     value="1.5" />
        </include>
    </group>

    <!-- start the rviz -->
    <node name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" args="-d $(find gpd_ros)/launch/gpd_pointnet_realsense.rviz" output="screen" />

	<node name="detect_grasps_pointnet" pkg="gpd_ros" type="detect_grasps_pointnet_realsense" output="screen">
 
       <!-- What type of point cloud is used and what ROS topic it comes from -->
       <param name="cloud_topic" value="/camera/depth/color/points" />

       <!-- Filepath to the configuration file for GPD -->
       <param name="config_file" value="$(find gpd_ros)/cfg/ros_gpd_pointnet_params.cfg" />

       <!-- Rviz plotting parameters -->
       <param name="rviz_topic" value="grasps_rviz" />

	</node>

</launch>
